packages/ng-query/src/lib/infinite-query.ts

```diff
@@ -15,135 +15,65 @@ import { buildQuery } from './utils';
   TQueryFnData = unknown,
-  TError = unknown,
+  TError = DefaultError,
   TData = TQueryFnData,
   TQueryData = TQueryFnData,
-  TQueryKey extends QueryKey = QueryKey
+  TQueryKey extends QueryKey = QueryKey,
+  TPageParam = unknown
 > = Unsubscribable & {
-  result$: Observable<InfiniteQueryObserverResult<TData, TError>>;
+  result$: Observable<
+    InfiniteQueryObserverResult<InfiniteData<TQueryData, TPageParam>, TError>
+  >;
 } & Omit<
     InfiniteQueryObserver<TQueryFnData, TError, TData, TQueryData, TQueryKey>,
     'subscribe'
   >;
+
 @Injectable({ providedIn: 'root' })
 export class InfiniteQueryService {
   private instance = inject(QueryClientService);
 
   use<
     TQueryFnData = unknown,
-    TError = unknown,
+    TError = DefaultError,
     TData = TQueryFnData,
-    TQueryKey extends QueryKey = QueryKey
+    TQueryKey extends QueryKey = QueryKey,
+    TPageParam = unknown
   >(
     options: NgQueryObserverOptions<
       TQueryFnData,
       TError,
       TData,
       TQueryFnData,
-      TQueryKey
-    >
-  ): NgInfiniteQueryObserverReturnType<TData, TError>;
-
-  use<
-    TQueryFnData = unknown,
-    TError = unknown,
-    TData = TQueryFnData,
-    TQueryKey extends QueryKey = QueryKey
-  >(
-    queryKey: TQueryKey,
-    options?: Omit<
-      NgQueryObserverOptions<
-        TQueryFnData,
-        TError,
-        TData,
-        TQueryFnData,
-        TQueryKey
-      >,
-      'queryKey'
-    >
-  ): NgInfiniteQueryObserverReturnType<TData, TError>;
-
-  use<
-    TQueryFnData = unknown,
-    TError = unknown,
-    TData = TQueryFnData,
-    TQueryKey extends QueryKey = QueryKey
-  >(
-    queryKey: TQueryKey,
-    queryFn: ObservableQueryFn<TQueryFnData, TQueryKey>,
-    options?: Omit<
-      NgQueryObserverOptions<
-        TQueryFnData,
-        TError,
-        TData,
-        TQueryFnData,
-        TQueryKey
-      >,
-      'queryKey' | 'queryFn'
-    >
-  ): NgInfiniteQueryObserverReturnType<TData, TError>;
-
-  use<
-    TQueryFnData,
-    TError,
-    TData = TQueryFnData,
-    TQueryKey extends QueryKey = QueryKey
-  >(
-    arg1:
-      | TQueryKey
-      | NgQueryObserverOptions<
-          TQueryFnData,
-          TError,
-          TData,
-          TQueryFnData,
-          TQueryKey
-        >,
-    arg2?:
-      | ObservableQueryFn<TQueryFnData, TQueryKey>
-      | NgQueryObserverOptions<
-          TQueryFnData,
-          TError,
-          TData,
-          TQueryFnData,
-          TQueryKey
-        >,
-    arg3?: NgQueryObserverOptions<
-      TQueryFnData,
-      TError,
-      TData,
-      TQueryFnData,
-      TQueryKey
+      TQueryKey,
+      TPageParam
     >
   ): NgInfiniteQueryObserverReturnType<TData, TError> {
-    const parsedOptions = parseQueryArgs(
-      arg1,
-      arg2 as any,
-      arg3
-    ) as QueryOptions;
-
     return buildQuery(
       this.instance,
       InfiniteQueryObserver as typeof QueryObserver,
-      parsedOptions
+      options as any
```

Comment:
@NetanelBasal  For now i typed this as `any` since i dont really know how to convert between them, maybe a bit help would be great.

Reply:
You still need multiple overloads. See for example the svelte version:
https://github.com/TanStack/query/blob/v5.4.2/packages/svelte-query/src/createQuery.ts